plugins/tpm-eventlog/fu-tpm-eventlog-device.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-common.h"
+
+#include "fu-tpm-eventlog-common.h"
+#include "fu-tpm-eventlog-device.h"
+
+struct _FuTpmEventlogDevice {
+	FuDevice		 parent_instance;
+	GPtrArray		*items;
+};
+
+G_DEFINE_TYPE (FuTpmEventlogDevice, fu_tpm_eventlog_device, FU_TYPE_DEVICE)
+
+typedef struct __attribute__ ((packed)) {
+	guint32		 pcr_index;
+	guint32		 event_type;
+	guint8		 digest[20];
+	guint32		 event_data_sz;
+} TpmPcrEvent;
+
+typedef struct __attribute__ ((packed)) {
+	guint32		 pcr_index;
+	guint32		 event_type;
+	guint32		 digestCount;
+} TpmPcrEvent2Header;
```

Comment:
These two come from somewhere exported?  If so, should try to include that header instead?

Reply:
Some defines are available, but alas not the header format structures. I've used what I can from tss2.